src/backend/distributed/utils/colocation_utils.c

```diff
@@ -1422,4 +1454,24 @@ EnsureTableCanBeColocatedWith(Oid relationId, char replicationModel,
 								  "%s and %s.", sourceRelationName,
 								  relationName)));
 	}
+
+	/* prevent colocating regular tables with tenant tables */
+	Oid targetRelationSchemaId = get_rel_namespace(relationId);
+	Oid sourceRelationSchemaId = get_rel_namespace(sourceRelationId);
+	if (IsTenantSchema(sourceRelationSchemaId) &&
+		targetRelationSchemaId != sourceRelationSchemaId)
+	{
+		char *relationName = get_rel_name(relationId);
+		char *sourceRelationName = get_rel_name(sourceRelationId);
+		char *sourceRelationSchemaName = get_namespace_name(sourceRelationSchemaId);
+
+		ereport(ERROR, (errmsg("cannot colocate tables %s and %s",
+							   sourceRelationName, relationName),
+						errdetail("Cannot colocate tables with tenant tables "
+								  "by using colocate_with option."),
+						errhint("Consider using \"ALTER TABLE %s SET SCHEMA %s;\" "
+								"to move the table to the same schema with the "
+								"tenant table.",
```

Comment:
is that possible?

Reply:
This is not possible today. But a time ago, I was thinking that we would definitely have such a logic for ALTER TABLE SET SCHEMA, in 12.0. Let me drop the hint and then we can re-add it if we decide implementing it in this release.